src/databroker_elasticsearch/brokersearch.py

```diff
@@ -44,10 +46,10 @@ def __call__(self, q, **kwargs):
         from elasticsearch.helpers import scan
         es = self.esindex.es
         index = self.esindex.index
-        doc_type = self.esindex.doc_type
-        query = {'q': q}
-        gscan = scan(es, query, index=index, doc_type=doc_type,
-                     _source=['uid'], **kwargs)
-        gstartstop = ((e['_source']['uid'], None) for e in gscan)
+        gscan = scan(es, q=q, index=index, _source=['uid'], **kwargs)
+        # FIXME: we need a fetchstart to work around bug in databroker.
+        # Remove the fetchstart mapping when fixed.
+        fetchstart = self.db.hs.mds.run_start_given_uid
+        gstartstop = ((fetchstart(e['_source']['uid']), None) for e in gscan)
```

Comment:
Would this work with `db[e['_source']['uid']]`? If the uid is not in the databroker we might want this to fall over, since that is weird behavior. This would get around from digging 3 layers deep into the databroker (esp. since this low level syntax might move when databroker moves to intake but the top level syntax is more likely to be stable).

Reply:
As noted the fetchstart call is temporary and is needed to work around https://github.com/NSLS-II/databroker/issues/397.
Once that bug is fixed, fetchstart can be dropped and Results would be instantiated from a sequence of `(start_uid, None)` pairs.

Using `db[uid]` is less efficient as each Header then needs to be created twice.
That is because `db[uid]` is a `Header` but search should return a `Results` type, so each
Header would need to be split to a start/stop documents and re-created as a new Header in [`Results.__iter__`](https://github.com/NSLS-II/databroker/blob/171c98543d7a06deee2d8518e54d4709653880b3/databroker/_core.py#L857-L862) again.